src/pybind/mgr/cephadm/serve.py

```diff
@@ -494,9 +497,16 @@ def matches_network(host):
                 daemon_type, daemon_id, host))
 
             try:
-                daemon_spec = self.mgr.cephadm_services[daemon_type].prepare_create(daemon_spec)
-                self.mgr._create_daemon(daemon_spec)
-                r = True
+                # HA_RGW needs to deploy 2 daemons
+                if daemon_type == 'HA_RGW':
+                    for dtype in ['haproxy', 'keepalived']:
+                        daemon_spec.daemon_type = dtype
```

Comment:
and this is the reason I really don't want to have two daemons for `ha-rgw`: we're introducing a whole mess of different `daemon_type`s and we'll have a hard time untangling them again.